### DISCUSSION TITLE
newline due to window size breaks link up for kitty

### DISCUSSION THREAD

**Discussion by k4lizen** [helix-editor/helix#11879] [General]
### Summary

![image](https://github.com/user-attachments/assets/c03202f1-5816-4038-bc35-f42eb8d5f6ec)
Not the whole link is underlined, and when I click it kitty takes me to only the part before the newline. I have
```toml
[editor]
soft-wrap.enable = true
```

I'm not sure whether helix is as fault here or kitty, but I wish clicking links like this worked.

### Reproduction Steps

 

I tried this:

1. put a long url in some file
2. open the file in kitty and helix
3. make the window shorter until the link is split on two lines
4. click the link

I expected this to happen:
I get taken to the full link.

Instead, this happened:
I only get taken to the part before the new line caused by the window width.

### Platform

linux

### Terminal Emulator

kitty

### Installation Method

source

### Helix Version

helix 24.3 (8a549b76)

**Comment by kirawi** [comment]
Use `gf` instead

**Reply by k4lizen** [reply]
Okay thats super useful and I will, thanks, but the bug still exists.

**Reply by kirawi** [reply]
See https://github.com/helix-editor/helix/issues/1472#issuecomment-1008644709

**Reply by k4lizen** [reply]
> I think we should support the hyperlink spec which makes terminal links clickable: https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda
> crossterm doesn't seem to support this, but termwiz does

So the bug is present because helix uses crossterm?

**Reply by kirawi** [reply]
It's not really a bug. In this case, it's just Kitty's algorithm for finding URLs not working well with terminal editors. The linked comment is about a spec that the upstream library we use for terminal manipulation would need to implement for us to control how terminal emulators interpret URLs.

**Reply by k4lizen** [reply]
> It's not really a bug.

The expected behaviour is clear, what is happening is not the expected behaviour and is detrimental to UX, it also occurs trivially. In my eyes that makes it a bug and should be tracked on the issue tracker. 

Seems theres an issue open on crossterm for it https://github.com/crossterm-rs/crossterm/issues/602 which is nice.
